subsys/net/ip/net_if.c

```diff
@@ -443,7 +443,8 @@ static void dad_timeout(struct k_work *work)
 	 */
 	tmp = net_if_ipv6_addr_lookup(&ifaddr->address.in6_addr, &iface);
 	if (tmp == ifaddr) {
-		net_mgmt_event_notify(NET_EVENT_IPV6_DAD_SUCCEED, iface);
+		net_mgmt_event_notify(NET_EVENT_IPV6_DAD_SUCCEED,
```

Comment:
Ẁe could skip all the changes to existing users of the net_mgmt_event_notify() if we have a separate function which has the info and length parameter. Then we can go through the listeners and convert them case by case separately.

Reply:
As noted previously, net_mgmt_event_notify() will be the with-info function. So changes will be required anyway for non-info based event notification (but yes, an inline one calling net_mgmt_event_notify() with NULL, 0 for info.)